### DISCUSSION TITLE
New editor environment variables not being reflected in builds

### DISCUSSION THREAD

**Discussion by ericmand** [codemagic-ci-cd/codemagic-docs#1648] [Q&A]
Most changes to my environment variables are not being reflected in builds (as evidenced by the dart-define flags that show up in the build command in the logs).

I had 5 env variables which worked. I added a 6th and 7th; saved; re-ran the build but only the original 5 were in the build command.

In order to get those two new ones included I wrote over two of the existing ones (that aren't critical), so there was still 5 variables and this change was only partially reflected in the build command: the old variable names were kept but values were blank (both of which weren't the intended outcome).

I think tried duplicating the pipeline, in case that addressed it. The new pipeline is behaving like the old one, injecting environment variables that don't reflect what is in the dashboard. 

The most recent build id is: 645d224f9d033b020a78b885

**Comment by icarusdust** [comment]
Hi @ericmand, I am not sure if I understood your query correctly and I see that your shared build is alright. You closed the discussion, assuming that you found an answer, if otherwise please let me know, so that I can continue looking into it.
